### PR TITLE
Add docker deploy recipe

### LINKED ISSUE #1667: Docker Deployment docs
I've noticed the docs prioritize Astro deployment with serverless (or static) hosts. I personally use Docker for deploying things on non-serverless hosts, so I feel others should be able to do the same without having to do any research.

I can easily make a tutorial and make a PR.

LMK if anyone has any questions!

### PR DESCRIPTION
**Pull request by ElianCodes**
#### What kind of changes does this PR include?

- New or updated content

#### Description

This adds the Docker page to the deployment recipes.

We should also change up the content of the GCP deployment a little since half of it is duplicated content. For now I think it's best to get this merge and then focus on making the GCP deployment guide compatible with the Docker guide.

- Closes #1667
- Replaces #2767

### CODE REVIEW

File: src/content/docs/en/recipes/docker.mdx [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+title: Build your Astro Site with Docker
+description: Learn how to build your Astro site using Docker.
+type: recipe
+i18nReady: false
+---
+
+[Docker](<https://docker.com>) is a tool to build, deploy, and run applications using containers.
+
+Docker images and containers can be deployed to many different platforms, like AWS, Azure, and [Google Cloud](</en/guides/deploy/google-cloud/#cloud-run-ssr-and-static>). This recipe won't cover how to deploy your site to a specific platform, but show you how to set up Docker for your project.
```
Comment (Yan-Thomas) [inline src/content/docs/en/recipes/docker.mdx:10]: Nit: unnecessary comma, also I personally find it weird to not have "will" here
```suggestion
Docker images and containers can be deployed to many different platforms, like AWS, Azure, and [Google Cloud](</en/guides/deploy/google-cloud/#cloud-run-ssr-and-static>). This recipe won't cover how to deploy your site to a specific platform but will show you how to set up Docker for your project.
```

File: src/content/docs/en/recipes/docker.mdx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+## Prerequisites
+
+- Docker installed on your local machine. You can find [installation instructions for your operating system here](<https://docs.docker.com/get-docker/>).
+- A Dockerfile in your project. You can [learn more about Dockerfiles here](https://docs.docker.com/engine/reference/builder/) and use the Dockerfiles in the following section as a starting point.
+
+## Creating a Dockerfile
+
+Create a file called `Dockerfile` in your project's root directory. This file contains the instructions to build your site, which will differ depending on your needs. This guide can't show all possible options, but will give you starting points for SSR and static mode.
```
Comment (Yan-Thomas) [inline src/content/docs/en/recipes/docker.mdx:19]: unnecessary comma
```suggestion
Create a file called `Dockerfile` in your project's root directory. This file contains the instructions to build your site, which will differ depending on your needs. This guide can't show all possible options but will give you starting points for SSR and static mode.
```

**Review by Yan-Thomas — COMMENTED** [review]
Thanks for the frequent amazing contributions @ElianCodes 💜, I have a couple nits, but overall LGTM

File: src/content/docs/en/recipes/docker.mdx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+### SSR
+
+This Dockerfile will build your site and serve it using Node.js on port `3000`. It requires the [Node adapter](/en/guides/integrations-guide/node/).
```
Comment (sarah11918) [inline src/content/docs/en/recipes/docker.mdx:23]:
```suggestion
This Dockerfile will build your site and serve it using Node.js on port `3000` and therefore requires the [Node adapter](/en/guides/integrations-guide/node/) installed in your Astro project.
```

File: src/content/docs/en/recipes/docker.mdx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+FROM httpd:2.4 AS runtime
+COPY --from=build /app/dist /usr/local/apache2/htdocs/
+EXPOSE 80
+```
+
+:::caution[Recommendation]
+Use this approach for simple websites that don't need any special configuration. For more complex websites, it is recommended to use a custom configuration, either in Apache or NGINX.
+:::
+
+#### NGINX
```
Comment (sarah11918) [inline src/content/docs/en/recipes/docker.mdx:71]: Is this the example of the more advanced configuration? I'm a little confused by the heading levels.

h3 - Static
h5 - Apache
h4 - NGINX  (And, I'm assuming everything down to `## Recipe` is all the NGINX stuff?

File: src/content/docs/en/recipes/docker.mdx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+RUN npm install
+COPY . .
+RUN npm run build -- --mode custom
+
+FROM nginx:alpine AS runtime
+COPY ./nginx/nginx.conf /etc/nginx/nginx.conf
+COPY --from=build /app/dist /usr/share/nginx/html
+```
+
+Before you can actually build the Dockerfile above, you'll need to create a configuration file for NGINX. Create a folder called `nginx` in your project's root directory and create a file called `nginx.conf` inside.
```
Comment (sarah11918) [inline src/content/docs/en/recipes/docker.mdx:86]:
```suggestion
In order to build the Dockerfile above, you'll also need to create a configuration file for NGINX. Create a folder called `nginx` in your project's root directory and create a file called `nginx.conf` inside.
```
 I'm not sure I like the "before  you can actually.. you need to"  phrasing. "Before" always SOUNDS like you were lied to (like, you should have done this before!) Even though that's not what you mean here. I might change to "in order to"...

**Review by sarah11918 — COMMENTED** [review]
Committed Yan's changes, and just left a couple of comments from me! Ready when you're happy you've addressed them!  🥳 

Is this our first non-Dan submitted recipe?????

**Comment by kevinzunigacuellar** [comment]
pinging myself 😅 (free hw)

File: src/content/docs/en/recipes/docker.mdx (hunk 4 above)
Comment (Jutanium) [reply src/content/docs/en/recipes/docker.mdx:71]: Yep, should be h4!

File: src/content/docs/en/recipes/docker.mdx [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+```docker title="Dockerfile" del={1} add={2}
+FROM node:lts as runtime
+FROM node:lts-alpine as runtime
+```
+:::
+
+### Static
+
+##### Apache (httpd)
```
Comment (Jutanium) [inline src/content/docs/en/recipes/docker.mdx:51]:
```suggestion
#### Apache (httpd)
```

**Pull request merged**